modin/engines/ray/pandas_on_ray/frame/partition.py

```diff
@@ -131,12 +139,21 @@ def drain_call_queue(self):
             return
         oid = self.oid
         call_queue = self.call_queue
-        (
-            self.oid,
-            self._length_cache,
-            self._width_cache,
-            self._ip_cache,
-        ) = deploy_ray_func.remote(call_queue, oid)
+        if len(call_queue) > 1:
```

Comment:
I agree with @vnlitvinov where he mentions above about this issue, it does seem a little weird. What is the benefit of special casing the call queue having length 1?

Reply:
This speeds up the operations that are map-like or include map phase.

**pandas_on_ray**

*BenchmarkMode.put(True)*

| df.shape = (8192, 1024) | df.round() | df.abs() | df.max() | df.prod() |
| - | - | - | - | - |
| master                           | 2.3758      | 2.0388  | 3.1617    | 4.0645    |
| PR                                 | 0.8098      | 0.7027  | 1.2626    | 1.6392    |

*BenchmarkMode.put(False)*

|df.shape = (8192, 1024)|df.round()|df.abs()|df.max()|df.prod()|
| - | - | - | - | - |
| master                           | 0.6013      | 0.5945  | 2.4051    | 3.2238    |
| PR                                 | 0.5044      | 0.4410  | 1.2138    | 1.2128    |

Reply:
Very interesting. Can we add a comment in the code to mention that it is written this way because it dramatically improves performance?

Reply:
Sure, will add.

Reply:
Added.

Reply:
@YarShev numbers look interesting... but can we see those comparing this PR against "this PR but without special handling of 1-long queue" to be sure we're not measuring some other effect?

Reply:
The numbers look even more interesting for the case without special handling of 1-long queue.

**pandas_on_ray**
*BenchmarkMode.put(True)*

| df.shape = (8192, 1024) | df.round() | df.prod() |
| - | - | - |
| master                           | 2.3758      | 4.0645    |
| PR (with the special casing)    | 0.8098      | 1.6392    |
| PR (with **no** the special casing)    | 3.2039      | 5.9272    |

*BenchmarkMode.put(False)*

| df.shape = (8192, 1024) | df.round() | df.prod() |
| - | - | - |
| master                           | 0.6013      | 3.2238    |
| PR (with the special casing)    | 0.5044      | 1.2128   |
| PR (with **no** the special casing)    | 0.8461      | 4.8586    |

It turns out if we pass `call_queue/funcs` in `apply_list_of_func` as second parameter, we will see that slowdown. And if we pass `call_queue/funcs` in `apply_list_of_func` as the first parameter, we will see numbers that are about equal to master. I am not sure of what is the reason but it will be fixed.

Reply:
Fixed